lib/puppet/configurer.rb

```diff
@@ -102,22 +102,31 @@ def retrieve_catalog(query_options)
   end
 
   # Convert a plain resource catalog into our full host catalog.
-  def convert_catalog(result, duration)
-    catalog = result.to_ral
-    catalog.finalize
-    catalog.retrieval_duration = duration
-    catalog.write_class_file
-    catalog.write_resource_file
+  def convert_catalog(result, duration, options = {})
+    catalog = nil
+
+    catalog_converstion_time = thinmark do
+      catalog = result.to_ral
+      catalog.finalize
+      catalog.retrieval_duration = duration
+      catalog.write_class_file
+      catalog.write_resource_file
+    end
+    options[:report].add_times(:convert_catalog, catalog_converstion_time) if options[:report]
```

Comment:
Since the `thinmark` logic is repeated, what about creating a helper method:

```ruby
def timed_block(report, metric, &block) do
  result = nil 
  time = thinmark do
    result = yield
  end
  report.add_times(metric, time) if report
  result
end
```

And then calling it from the various places:

```ruby
catalog = timed_block(options[:report], :convert_catalog) do
  catalog = result.to_ral
  catalog.write_resource_file
  catalog.finalize
  catalog.retrieval_duration = duration
  catalog.write_class_file
  catalog.write_resource_file
  catalog
end

Reply:
There are only two locations where this would be needed, so I'm not sure it's worth it quite yet. I can still do this if you think it's worth it though

Reply:
ah no worries, that's fine